var/spack/repos/builtin/packages/lemon/package.py

```diff
@@ -16,3 +16,49 @@ class Lemon(CMakePackage):
     url = "https://lemon.cs.elte.hu/pub/sources/lemon-1.3.1.tar.gz"
 
     version("1.3.1", sha256="71b7c725f4c0b4a8ccb92eb87b208701586cf7a96156ebd821ca3ed855bad3c8")
+
+    variant("coin", default=False, description="Enable Coin solver backend")
+    variant("ilog", default=False, description="Enable ILOG (CPLEX) solver backend")
+    variant("glpk", default=True, description="Enable GLPK solver backend")
+    # soplex not mentioned in docs but shown in cmakecache
+    variant("soplex", default=False, description="Enable SOPLEX solver backend")
+
+    variant(
+        "lp_backend",
+        default="glpk",
+        description="Set default LP solver backend",
+        values=("glpk", "cplex", "clp", "soplex"),
+        multi=False,
+    )
+    variant(
+        "mip_backend",
+        default="glpk",
+        description="Set default MIP solver backend",
+        values=("glpk", "cplex", "cbc"),
+        multi=False,
+    )
+
+    depends_on("glpk", when="+glpk")
+    # LP solver
+    depends_on("glpk", when="lp_backend=glpk")
+    depends_on("cplex", when="lp_backend=cplex")
+    depends_on("clp", when="lp_backend=clp")
+    # MIP solver
+    depends_on("glpk", when="mip_backend=glpk")
+    depends_on("cplex", when="mip_backend=cplex")
+    depends_on("cbc", when="mip_backend=cbc")
```

Comment:
Does the software build support any combination of `glpk`, `lp_backend`, and `mip_backend`?  

If only one can be provided, then you could use `conflict()` directives (see https://spack.readthedocs.io/en/latest/packaging_guide.html#conflicts-and-requirements).

Reply:
removed these. According to cmake files these're automatically set.
```
IF(ILOG_FOUND)
  SET(DEFAULT_LP "CPLEX")
  SET(DEFAULT_MIP "CPLEX")
ELSEIF(COIN_FOUND)
  SET(DEFAULT_LP "CLP")
  SET(DEFAULT_MIP "CBC")
ELSEIF(GLPK_FOUND)
  SET(DEFAULT_LP "GLPK")
  SET(DEFAULT_MIP "GLPK")
ELSEIF(SOPLEX_FOUND)
  SET(DEFAULT_LP "SOPLEX")
ENDIF()
```